torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -31,6 +32,11 @@ void addSendRpcBackward(
     autogradContext.addSendFunction(
         grad_fn, autogradMetadata.autogradMessageId);
   }
+
+  // Record the workerID
+  if (dst.id_ != -1) {
```

Comment:
We should record the workerId only if we are attaching a send function above. If we don't attach a send function, its likely there isn't a autograd context to clean up on dst.